app/src/main/java/me/martinez/sergio/daggertwobasearchitecture/di/ComponentProxy.java

```diff
@@ -0,0 +1,25 @@
+package me.martinez.sergio.daggertwobasearchitecture.di;
+
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivity;
+import me.martinez.sergio.daggertwobasearchitecture.activities.MainActivityComponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.AppComponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.SectionComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragmentComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.firststep.FirstFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.firststep.FirstFragmentComponent;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.secondstep.SecondFragment;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.sections.testsection.secondstep.SecondFragmentComponent;
+
+/**
+ * This interface sits between the dagger component implementations and its usages. This way,
+ * a certain injection is not coupled with the component-module hierarchy.
+ */
+public interface ComponentProxy {
```

Comment:
Exactly, one of the key points here is to make our business logic, or activity, fragment code (which are not supposed to have business logic, but that's another story) abstract and not to care about a certain module, component hierarchy.